src/org/labkey/test/pages/core/login/LoginConfigRow.java

```diff
@@ -0,0 +1,115 @@
+package org.labkey.test.pages.core.login;
+
+import org.labkey.test.Locator;
+import org.labkey.test.components.WebDriverComponent;
+import org.labkey.test.params.login.AuthenticationProvider;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+
+public class LoginConfigRow extends WebDriverComponent<LoginConfigRow.ElementCache>
+{
+    final WebElement _el;
+    final WebDriver _driver;
+
+    public LoginConfigRow(WebElement element, WebDriver driver)
+    {
+        _el = element;
+        _driver = driver;
+    }
+
+    public String getDescription()
+    {
+        return elementCache().description.getText();
+    }
+
+    public String getDetail()
+    {
+        return elementCache().details.getText();
+    }
+
+    public String getProvider()
+    {
+        return elementCache().provider.getText();
+    }
+
+    public LoginConfigurePage clickDelete()
+    {
+        elementCache().deleteButton.click();
+        getWrapper().acceptAlert();
+        getWrapper().shortWait().until(ExpectedConditions.stalenessOf(getComponentElement()));
+        return new LoginConfigurePage(getDriver());
+    }
+
+    public <P extends AuthDialogBase> P clickEdit(AuthenticationProvider<P> authenticationProvider)
+    {
+        elementCache().editButton.click();
+        return authenticationProvider.getEditDialog(this);
+    }
+
+    @Override
+    public WebElement getComponentElement()
+    {
+        return _el;
+    }
+
+    @Override
+    public WebDriver getDriver()
+    {
+        return _driver;
+    }
+
+    @Override
+    protected ElementCache newElementCache()
+    {
+        return new ElementCache();
+    }
+
+
+    protected class ElementCache extends WebDriverComponent.ElementCache
+    {
+        final WebElement baseFieldsElement = Locator.tagWithClass("div", "domain-row-base-fields").findWhenNeeded(this);
+        final WebElement description = Locator.tagWithClass("div", "description").findElement(baseFieldsElement);
+        final WebElement details = Locator.tagWithClass("div", "details").findElement(baseFieldsElement);
+        final WebElement provider = Locator.tagWithClass("div", "provider").findElement(baseFieldsElement);
```

Comment:
Any reason to not use `findWhenNeeded` for these?

Reply:
nope.  it's a good idea